cashu/core/base.py

```diff
@@ -115,6 +115,15 @@ class P2PKWitness(BaseModel):
     def from_witness(cls, witness: str):
         return cls(**json.loads(witness))
 
+class DLCWitness(BaseModel):
+    leaf_secret: str
+    merkle_proof: List[str]
+    witness: Optional[str] = None
+
+    @classmethod
+    def from_witness(cls, witness: str):
+        return cls(**json.loads(witness))
```

Comment:
I noticed this conflicts with the spec (the witness in the spec was an object, not a string). I see `Proof.witness` is strictly typed to be an `Optional[str]` in nutshell, so i changed the spec to match. 

https://github.com/cashubtc/nuts/pull/128/commits/d1ffefa8d8eb15aac55468948d9075e09ab83494